content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,262 @@ The following are not considered as part of the adversarial model:
 In specific, the communication channels are assumed to be secure.
 
 # Wire Specification
-Peers communicate with each other using a request / response API. 
-The messages sent are Protobuf RPC messages which are implemented using [protocol buffers v3](https://developers.google.com/protocol-buffers/).
-The following are the specifications of the Protobuf messages.
 
 ## Payloads
 
 ```protobuf
 syntax = "proto3";
 
-message Index {
-  bytes digest = 1;
-  sint64 receiverTime = 2;
-  sint64 senderTime = 3;
-  string pubsubTopic = 4;
-}
-
-message PagingInfo {
-  uint64 pageSize = 1;
-  Index cursor = 2;
-  enum Direction {
-    BACKWARD = 0;
-    FORWARD = 1;
-  }
-  Direction direction = 3;
-}
+// Protocol identifier: /vac/waku/store-query/3.0.0
+package waku.store.v3;
 
-message ContentFilter {
-  string contentTopic = 1;
-}
+import "waku/message/v1/message.proto";
 
-message HistoryQuery {
-  // the first field is reserved for future use
-  string pubsubtopic = 2;
-  repeated ContentFilter contentFilters = 3;
-  PagingInfo pagingInfo = 4;
+message WakuMessageKeyValue {
+  optional bytes message_hash = 1; // Globally unique key for a Waku Message
+  optional waku.message.v1.WakuMessage message = 2; // Full message content as value
 }
 
-message HistoryResponse {
-  // the first field is reserved for future use
-  repeated WakuMessage messages = 2;
-  PagingInfo pagingInfo = 3;
-  enum Error {
-    NONE = 0;
-    INVALID_CURSOR = 1;
-  }
-  Error error = 4;
+message StoreQueryRequest {
+  string request_id = 1;
+  bool include_data = 2; // Response should include full message content
+  
+  // Filter criteria for content-filtered queries
+  optional string pubsub_topic = 10;
+  repeated string content_topics = 11;
+  optional int64 time_start = 12;
+  optional int64 time_end = 13;
```

Comment:
According to protobuffer docs, `int` are  `Inefficient for encoding negative numbers – if your field is likely to have negative values, use sint64 instead`. 

```suggestion
  optional sint64 time_start = 12;
  optional sint64 time_end = 13;
```

Reply:
Note I'm not suggesting here to use `uint` because we use `int64` in the nwaku code for dealing with timestamps (also unix time functions across languages generally use `int64`)

Reply:
Thanks for this. It's indeed an oversight - we decided in Store v2 already to use `zint` for timestamps. Afaik the nwaku implementation for Store v3 should already be using `zint`, so it was only this proto that needed changing.